src/stdlib_logger.f90

```diff
@@ -544,27 +544,33 @@ subroutine format_output_string( self, unit, string, procedure_name, &
 
         subroutine format_first_line()
 
-            if ( length <= self % max_width .or. self % max_width == 0 ) then
+            if ( self % max_width == 0 .or. &
+                ( length <= self % max_width .and. &
+                index( string(1:min(length, self % max_width)), new_line('a')) == 0 &
+                )) then
```

Comment:
As `length <= self % max_width`, `string(1:min(length, self % max_width))` is equivalent to `string(1:length)`.

Reply:
Thank you. I changed it as suggested.